zerver/lib/zcommand.py

```diff
@@ -73,4 +77,52 @@ def change_mode_setting(
                 setting_value=False,
             )
         )
+    elif command == "digress":
+        data_keys = [
+            "old_stream",
+            "old_topic",
+            "new_stream",
+            "new_topic",
+        ]
+        if not data or not all(key in data for key in data_keys):
+            raise JsonableError(_("Invalid data."))
```

Comment:
Somewhere in here we should validate that `new_topic` is a valid topic (not too long, has been stripped), ideally reusing whatever helper we have from the `send_message` code path.  And make sure we test for that case.

(I think it'll currently fail in the `internal_send_message` below, but we possibly after sending one but not the other of the two messages, which is clearly no good).

Reply:
Done, I added a call to `truncate_topic` to clean the new topic.